### PR TITLE
Alter index is not supported for vector index

### PR DESCRIPTION
**Pull request by azevaykin**
### Changelog entry 

Disable ALTER INDEX for vector index
https://ydb.tech/docs/en/yql/reference/syntax/alter_table#alter-index

### Changelog category 

* Not for changelog (changelog entry is not required)

### Additional information

...

### CODE REVIEW

**Review by MBkkt — COMMENTED** [review]
Maybe you can add some tests for describe vector index?

Because it looks like rpc_describe didn't support vector index before this changes

**Review by MBkkt — DISMISSED** [review]
Test in the second PR

**Comment by azevaykin** [comment]
> Maybe you can add some tests for describe vector index?
> 
> Because it looks like rpc_describe didn't support vector index before this changes

fa955aba25eb78926cc6f28a7cb08e2774389c37

**Review by MBkkt — DISMISSED** [review]

File: ydb/core/kqp/provider/yql_kikimr_exec.cpp [hunk 1]
```diff
@@ -1682,9 +1682,24 @@ class TKiSinkCallableExecutionTransformer : public TAsyncCallbackTransformer<TKi
...
+                            const auto indexName = indexSetting.Value().Cast<TCoAtom>().StringValue();
+                            const auto indexIter = std::find_if(table.Metadata->Indexes.begin(), table.Metadata->Indexes.end(), [&indexName] (const auto& index) {
+                                return index.Name == indexName;
+                            });
+                            if (!indexIter) {
+                                ctx.AddError(
+                                    TIssue(ctx.GetPosition(indexSetting.Name().Pos()),
+                                            TStringBuilder() << "Unknown index name: " << indexName));
+                                return SyncError();
+                            }
```
Comment (MBkkt) [inline ydb/core/kqp/provider/yql_kikimr_exec.cpp:1694]: Where is this tested?

Comment (azevaykin) [reply ydb/core/kqp/provider/yql_kikimr_exec.cpp:1694]: Generic error here:
https://github.com/ydb-platform/ydb/pull/7005/files#diff-b80980f1827cf5f57676941f35365f578fb3063b9ef47370835a1af8be6a1dfeR2512

Comment (azevaykin) [reply ydb/core/kqp/provider/yql_kikimr_exec.cpp]: I have added new test [AlterTableAlterMissedIndex](https://github.com/ydb-platform/ydb/pull/7005/files#diff-b80980f1827cf5f57676941f35365f578fb3063b9ef47370835a1af8be6a1dfeR2514)

**Review by MBkkt — APPROVED** [review]

**Review by gridnevvvit — APPROVED** [review]

**Pull request merged**